staging/src/k8s.io/client-go/util/certificate/csr/csr.go

```diff
@@ -307,6 +310,8 @@ func ensureCompatible(new, orig *certificatesv1.CertificateSigningRequest, priva
 	if len(new.Spec.SignerName) > 0 && len(orig.Spec.SignerName) > 0 && new.Spec.SignerName != orig.Spec.SignerName {
 		return fmt.Errorf("csr signerNames differ: new %q, orig: %q", new.Spec.SignerName, orig.Spec.SignerName)
 	}
+	// TODO do we need a check for durationHint here?
```

Comment:
we definitely have to tolerate the field not being present to avoid breaking against old servers. given the signer doesn't have to honor the hint, I would tend not to include it in our "should I start a new CSR" logic